src/hotspot/share/opto/node.cpp

```diff
@@ -2683,46 +2683,52 @@ void Node::dump_comp(const char* suffix, outputStream *st) const {
 // For each input edge to a node (ie - for each Use-Def edge), verify that
 // there is a corresponding Def-Use edge.
 //------------------------------verify_edges-----------------------------------
-void Node::verify_edges(Unique_Node_List &visited) {
-  uint i, j, idx;
-  int  cnt;
-  Node *n;
+void Node::verify_edges(Node* root, Unique_Node_List &visited, Node_List &nstack) {
+  nstack.push(root);
 
-  // Recursive termination test
-  if (visited.member(this))  return;
-  visited.push(this);
-
-  // Walk over all input edges, checking for correspondence
-  for( i = 0; i < len(); i++ ) {
-    n = in(i);
-    if (n != NULL && !n->is_top()) {
-      // Count instances of (Node *)this
-      cnt = 0;
-      for (idx = 0; idx < n->_outcnt; idx++ ) {
-        if (n->_out[idx] == (Node *)this)  cnt++;
-      }
-      assert( cnt > 0,"Failed to find Def-Use edge." );
-      // Check for duplicate edges
-      // walk the input array downcounting the input edges to n
-      for( j = 0; j < len(); j++ ) {
-        if( in(j) == n ) cnt--;
+  while (nstack.size() > 0) {
+    Node* next = nstack.pop();
+    if (visited.member(next)) {
+      continue;
+    }
+    visited.push(next);
+
+    // Walk over all input edges, checking for correspondence
+    uint length = next->len();
+    for (uint i = 0; i < length; i++) {
+      Node* n = next->in(i);
```

Comment:
I suggest to rename `n` to `input` to make it easier to see if it is the current node or an input to it.

Reply:
Renamed to `in`